src/app/qgsdecorationnortharrow.cpp

```diff
@@ -94,128 +97,145 @@ void QgsDecorationNorthArrow::run()
+                                   ) - centerYDouble );
+    // need width/height of paint device
+    int deviceHeight = context.painter()->device()->height();
+    int deviceWidth = context.painter()->device()->width();
+
+    // Set  margin according to selected units
+    int xOffset = 0;
+    int yOffset = 0;
+    switch ( mMarginUnit )
+    {
+      case QgsUnitTypes::RenderMillimeters:
+      {
+        int pixelsInchX = context.painter()->device()->logicalDpiX();
+        int pixelsInchY = context.painter()->device()->logicalDpiY();
+        xOffset = pixelsInchX * INCHES_TO_MM * mMarginHorizontal;
+        yOffset = pixelsInchY * INCHES_TO_MM * mMarginVertical;
+        break;
+      }
 
-      //unrotate the canvas again
-      context.painter()->restore();
+      case QgsUnitTypes::RenderPixels:
+        xOffset = mMarginHorizontal - 5; // Minus 5 to shift tight into corner
+        yOffset = mMarginVertical - 5;
+        break;
+
+      case QgsUnitTypes::RenderPercentage:
+        xOffset = ( ( deviceWidth - size.width() ) / 100. ) * mMarginHorizontal;
+        yOffset = ( ( deviceHeight - size.width() ) / 100. ) * mMarginVertical;
+        break;
+      case QgsUnitTypes::RenderMapUnits:
+      case QgsUnitTypes::RenderPoints:
+      case QgsUnitTypes::RenderInches:
+      case QgsUnitTypes::RenderUnknownUnit:
+      case QgsUnitTypes::RenderMetersInMapUnits:
+        break;
     }
-    else
+    //Determine placement of label from form combo box
+    switch ( mPlacement )
     {
-      QFont myQFont( QStringLiteral( "time" ), 12, QFont::Bold );
-      context.painter()->setFont( myQFont );
-      context.painter()->setPen( Qt::black );
-      context.painter()->drawText( 10, 20, tr( "North arrow pixmap not found" ) );
+      case BottomLeft:
+        context.painter()->translate( xOffset, deviceHeight - yOffset - maxLength + ( maxLength - size.height() ) / 2 );
+        break;
+      case TopLeft:
+        context.painter()->translate( xOffset, yOffset );
+        break;
+      case TopRight:
+        context.painter()->translate( deviceWidth - xOffset - maxLength + ( maxLength - size.width() ) / 2, yOffset );
+        break;
+      case BottomRight:
+        context.painter()->translate( deviceWidth - xOffset - maxLength + ( maxLength - size.width() ) / 2,
+                                      deviceHeight - yOffset - maxLength + ( maxLength - size.height() ) / 2 );
+        break;
     }
+
+    //rotate the canvas by the north arrow rotation amount
+    context.painter()->rotate( mRotationInt );
+    //Now we can actually do the drawing, and draw a smooth north arrow even when rotated
+    context.painter()->translate( xShift, yShift );
+    svg.render( context.painter(), QRectF( 0, 0, size.width(), size.height() ) );
+
+    //unrotate the canvas again
+    context.painter()->restore();
+  }
+  else
+  {
+    QFont font( QStringLiteral( "time" ), 12, QFont::Bold );
```

Comment:
I've never heard of the "time" font before!

Reply:
@nyalldawson , since the default SVG is embedded into QGIS binaries (since 3.0), I'd be in favor of removing that altogether. Agreed?